src/detect-pcre.c

```diff
@@ -918,6 +918,9 @@ static int DetectPcreSetup (DetectEngineCtx *de_ctx, Signature *s, const char *r
     if (sm_list == -1)
         goto error;
 
+    if ((sm_list == DETECT_SM_LIST_BASE64_DATA) && (s->base64_decoded_cnt > 0)) {
```

Comment:
why is this is necessary?

Reply:
Because I found some rules in ET ruleset where `base64_data` was followed by `pcre` keyword and no content. It is mostly to make my debug validation check happy (e.g. sid `2035884`) which is to ensure that all base64 data is actually set up to match corresponding to some `content` or something (`pcre` in this case).
Q: Do you think this makes sense?